tube/etl/indexers/injection/new_translator.py

```diff
@@ -227,18 +231,9 @@ def get_aggregating_props(self):
                     props.append(self.clone_prop_with_iterator_fn(p))
         return props
 
-    def translate_final(self):
-        """
-        Because one file can belong to multiple root nodes (case, subject).
-        In the final step of file document, we must construct the list of root instance's id
-        :return:
-        """
-        df = self.load_from_hadoop_to_dateframe()
-        aggregating_props = self.get_aggregating_props()
-        if len(aggregating_props) == 0:
-            return df
-
+    def flatten_nested_list(self, df):
         expr = []
+        aggregating_props = self.get_aggregating_props()
```

Comment:
`translate_final()` could pass `aggregating_props` to `flatten_nested_list()` in a parameter so we don't have to rerun `self.get_aggregating_props()`